File: Julia/8kyu/8kyu-the-wide-mouthed-frog.jl
# The Wide-Mouthed frog!
# Language: Julia
# Version: 1.5
# Difficulty: 8 kyu

# Solution 1 (Julia specific)
function mouthsize(animal)
    lowercase(animal) == "alligator" ? "small" : "wide"
end

# Solution 2 (similar to R)
function mouthsize(animal)
    ifelse(lowercase(animal) == "alligator", "small", "wide")
  end


# Tests
using FactCheck

facts("Basic tests") do
  context("Should pass fixed tests") do
    @fact mouthsize("toucan") --> "wide"
    @fact mouthsize("ant bear") --> "wide"
    @fact mouthsize("alligator") --> "small"
  end
end

File: R/8kyu/8kyu-the-wide-mouthed-frog.r
# The Wide-Mouthed frog!
# Language: R
# Version 3.4.1
# Difficulty: 8 kyu

mouth_size <- function(animal){
  ifelse(tolower(animal) == "alligator", "small", "wide")
}


# Tests
test_that('Basic tests', {
  expect_equal(mouth_size("toucan"),"wide")
  expect_equal(mouth_size("ant bear"),"wide")
  expect_equal(mouth_size("alligator"),"small")
})